js/packages/openinference-instrumentation-openai/test/openai.test.ts

```diff
@@ -421,34 +422,273 @@ describe("OpenAIInstrumentation", () => {
-  "llm.input_messages.0.message.content": "How is the weather this week?",
+  "input.value": "{"messages":[{"role":"user","content":"What's the weather today?"}],"model":"gpt-3.5-turbo","tools":[{"type":"function","function":{"name":"getCurrentLocation","parameters":{"type":"object","properties":{}},"description":"Get the current location of the user."}},{"type":"function","function":{"name":"getWeather","description":"Get the weather for a location.","parameters":{"type":"object","properties":{"location":{"type":"string"}}}}}],"stream":true}",
+  "llm.input_messages.0.message.content": "What's the weather today?",
   "llm.input_messages.0.message.role": "user",
-  "llm.input_messages.1.message.role": "assistant",
-  "llm.input_messages.1.message.tool_calls.0.tool_call.function.arguments": "{}",
-  "llm.input_messages.1.message.tool_calls.0.tool_call.function.name": "getCurrentLocation",
-  "llm.input_messages.2.message.content": "Boston",
-  "llm.input_messages.2.message.role": "tool",
-  "llm.input_messages.3.message.role": "assistant",
-  "llm.input_messages.3.message.tool_calls.0.tool_call.function.arguments": "{
-  "location": "Boston"
-}",
-  "llm.input_messages.3.message.tool_calls.0.tool_call.function.name": "getWeather",
-  "llm.input_messages.4.message.content": "{"temperature":52,"precipitation":"rainy"}",
-  "llm.input_messages.4.message.role": "tool",
-  "llm.invocation_parameters": "{"model":"gpt-3.5-turbo","tools":[{"type":"function","function":{"name":"getCurrentLocation","parameters":{"type":"object","properties":{}},"description":"Get the current location of the user."}},{"type":"function","function":{"name":"getWeather","parameters":{"type":"object","properties":{"location":{"type":"string"}}},"description":"Get the weather for a location."}}],"tool_choice":"auto","stream":false}",
-  "llm.model_name": "gpt-3.5-turbo-0613",
-  "llm.output_messages.0.message.content": "The weather in Boston this week is expected to be rainy with a temperature of 52 degrees.",
+  "llm.invocation_parameters": "{"model":"gpt-3.5-turbo","tools":[{"type":"function","function":{"name":"getCurrentLocation","parameters":{"type":"object","properties":{}},"description":"Get the current location of the user."}},{"type":"function","function":{"name":"getWeather","description":"Get the weather for a location.","parameters":{"type":"object","properties":{"location":{"type":"string"}}}}}],"stream":true}",
+  "llm.model_name": "gpt-3.5-turbo",
+  "llm.output_messages.0.message.content": "",
   "llm.output_messages.0.message.role": "assistant",
```

Comment:
Role is not correct here, there are specialized "tool_call" and "function_call" roles.

Reply:
This is not the tool call but the assistant saying invoke a tool

```
// {role: "user",      content: "How's the weather this week?"}
// {role: "assistant", tool_calls: [{type: "function", function: {name: "getCurrentLocation", arguments: "{}"}, id: "123"}
// {role: "tool",      name: "getCurrentLocation", content: "Boston", tool_call_id: "123"}
// {role: "assistant", tool_calls: [{type: "function", function: {name: "getWeather", arguments: '{"location": "Boston"}'}, id: "1234"}]}
// {role: "tool",      name: "getWeather", content: '{"temperature": "50degF", "preciptation": "high"}', tool_call_id: "1234"}
// {role: "assistant", content: "It's looking cold and rainy - you might want to wear a jacket!"}
//
// Final content: "It's looking cold and rainy - you might want to wear a jacket!"
```